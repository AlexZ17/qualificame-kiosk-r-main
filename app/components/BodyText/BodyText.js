import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './BodyTextStyle';
import { Fonts, Colors } from 'qualificame-kiosk/app/styles';

export default class BodyText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		
		return (
			<View>
				<Text style={[styles.bodyText, textColor, weight, align]}>
					{this.props.text}
				</Text>
			</View>
		);
	}
}

	BodyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	BodyText.defaultProps = {
		text: '',
		color: "black",
		weight: "regular",
		align: "left"
	}
