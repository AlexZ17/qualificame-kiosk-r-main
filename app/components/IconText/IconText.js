import React, { PureComponent } from 'react';
import { Text, View, Platform } from 'react-native';
import PropTypes from 'prop-types';

import styles from './IconTextStyle';
import SubtitleText from 'qualificame-kiosk/app/components/SubtitleText';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-kiosk/app/styles';

export default class IconText extends PureComponent {
	render() {
		let color = {color: Colors[this.props.color]}
		return (
			<View style={[styles.row, styles.alignItemsCenter, (this.props.AddVerticalMargin && styles.baseVerticalMargin), (this.props.AddHorizontalMargin && styles.baseHorizontalMargin)]}>
				<Feather
					name={this.props.iconName} 
					size={Platform.OS === 'ios' ? 29:24} 
					style={[color]}
				/>
				<View style={styles.smallHorizontalMargin}>
					<SubtitleText
						text={this.props.text}
						color={this.props.color}
					/>
				</View>
			</View>
		);
	}
}

	IconText.propTypes = {
		iconName: PropTypes.string,
		text: PropTypes.string,
		color: PropTypes.string,
		AddVerticalMargin: PropTypes.bool,
		AddHorizontalMargin: PropTypes.bool,
	}

	IconText.defaultProps = {
		iconName: 'phone',
		text: '',
		color: 'white',
		AddVerticalMargin: false,
		AddHorizontalMargin: false,
	}
