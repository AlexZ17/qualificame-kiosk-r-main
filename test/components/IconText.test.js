import React from 'react';
import { shallow } from 'enzyme';

import IconText from 'qualificame-kiosk/app/components/IconText';
import { Feather } from '@expo/vector-icons';
import BodyText from 'qualificame-kiosk/app/components/BodyText';

function createTestProps (props) {
	return {
		// common props
		color: 'red',
		text: 'testing',
		iconName: 'moon',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<IconText {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders Feather component', () => {
		expect(wrapper.find(Feather)).toHaveLength(1);
	});
	it('renders BodyText component', () => {
		expect(wrapper.find(BodyText)).toHaveLength(1);
	});
	it('displays the correct feather icon ', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
		expect(wrapper.find(Feather).props().name).toEqual('moon')
	});
	it('renders text correctly', () => {
		expect(wrapper.find(BodyText).prop('text')).toEqual('testing');
	});
})