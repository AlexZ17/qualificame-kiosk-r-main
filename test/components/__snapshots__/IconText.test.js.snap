// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      false,
      false,
    ]
  }
>
  <Icon
    allowFontScaling={false}
    name="moon"
    size={29}
    style={
      Array [
        Object {
          "color": "#EB3F3F",
        },
      ]
    }
  />
  <Component
    style={
      Object {
        "marginHorizontal": 8,
      }
    }
  >
    <BodyText
      align=""
      color="red"
      text="testing"
      weight=""
    />
  </Component>
</Component>
`;
